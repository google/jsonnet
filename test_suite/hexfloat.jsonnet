/*
Copyright 2018 NVIDIA Corporation. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

{
  a: 0,
  b: 0x0,
  c: 0x1,
  d: 0x2A,
  e: 0x3a,
  f: 0x4E,
  g: 0x5e,
  h: 0x6.8,
  i: 0x7.81,
  j: 0x8.8A,
  k: 0x9.8a,
  l: 0xa.8E,
  m: 0xE.8e,
  n: 0xfcp2,
  o: 0xAdp+4,
  p: 0xbep-8,
  q: 0xBf.2p2,
  r: 0xcA.2p+4,
  s: 0xDB.2p-8,
  t: 0xdC.2p10,
  u: 0xDD.2p+11,
  v: 0xFE.2p-12,
  w: 0xFF.2p01,
  x: 0xF.F2p+01,
  y: 0x.FF2p-01,
}
