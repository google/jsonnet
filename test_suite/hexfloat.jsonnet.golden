{
   "a": 0,
   "b": 0,
   "c": 1,
   "d": 42,
   "e": 58,
   "f": 78,
   "g": 94,
   "h": 6.5,
   "i": 7.50390625,
   "j": 8.5390625,
   "k": 9.5390625,
   "l": 10.5546875,
   "m": 14.5546875,
   "n": 1008,
   "o": 2768,
   "p": 0.7421875,
   "q": 764.5,
   "r": 3234,
   "s": 0.85595703125,
   "t": 225408,
   "u": 452864,
   "v": 0.062042236328125,
   "w": 510.25,
   "x": 31.890625,
   "y": 0.498291015625
}
